code/script.js

```diff
@@ -244,100 +245,140 @@ const generateBoard = () => {
+    }
+  } 
 
-// It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
-      )
+  const category = currentQuestion.category
+  const attribute = currentQuestion.attribute
+  const value = currentQuestion.value
+
+  if (category === 'hair color'){
+    if (keep){
+      alert(`Yes, the person has ${value} hair! Keep all people with ${value} hair`)
     } else {
-      alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
-      )
+      alert(`No, the person doesn't have ${value} hair! Remove all people with ${value} hair`)
     }
-  } else if (group === 'other') {
-    // Similar to the one above
-  } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+  } else if (category === 'eye color'){
+    if (keep){
+      alert(`Yes, the person has ${value} eyes! Keep all people with ${value} eyes`)
+    } else {
+      alert(`No, the person doesn't have ${value} eyes! Remove all people with ${value} eyes`)
+    }
+  } else if (category === 'accessories'){
+    if (keep){
+      alert(`Yes, the person wears ${attribute}! Keep all people with ${attribute}`)
+    } else {
+      alert(`No, the person doesn't wear ${attribute}! Remove all people with ${attribute}`)
+    }
+  } else if (category === 'other'){
+    if (keep){
+      alert(`Yes, the person is a ${attribute}! Keep all smokers`)
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      alert(`No, the person isn't a ${attribute}! Remove all smokers`)
     }
   }
 
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
-  // Invoke a function to redraw the board with the remaining people.
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  }
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  let confirmMessage = (`Do you really want to make a guess on ${suspect}?`)
+  let confirmGuess = confirm(confirmMessage)
+  if (confirmGuess === true) {
+    checkMyGuess(suspect)
+  } else {
+    alert(`OK! Keep on guessing`)
+  }
```

Comment:
Is this necessary? because I never see this alert when playing the game?

Reply:
It is shown when you type guess and then you do not confirm it 😸